SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -1380,15 +1380,15 @@ void USpatialActorChannel::ClientProcessOwnershipChange(bool bNewNetOwned)
 	}
 }
 
-void USpatialActorChannel::OnSubobjectDeleted(const FUnrealObjectRef& ObjectRef, UObject* Object)
+void USpatialActorChannel::OnSubobjectDeleted(const FUnrealObjectRef& ObjectRef, UObject* Object, const TWeakObjectPtr<UObject>& ObjectWeakPtr)
 {
 	CreateSubObjects.Remove(Object);
-
+	
 	Receiver->MoveMappedObjectToUnmapped(ObjectRef);
-	if (FSpatialObjectRepState* SubObjectRefMap = ObjectReferenceMap.Find(Object))
+	if (FSpatialObjectRepState* SubObjectRefMap = ObjectReferenceMap.Find(ObjectWeakPtr))
```

Comment:
If ObjectWeakPtr.IsValid() is false, then this will find any invalid weak object pointer in the map, won't it?

Reply:
(see FWeakObjectPtr::operator==)

Reply:
Agree with Tim - Epic discourages weak pointers as keys, see here under "Usage Warnings":
https://docs.unrealengine.com/en-US/Programming/UnrealArchitecture/SmartPointerLibrary/WeakPointer/index.html

FObjectKey is worth considering instead as a key for the TMaps.

Reply:
TWeakObjectPtrMapKeyFuncs would sidestep that issue.